### PR TITLE
#2527: Defaults to filter statuses based only on current

### PR DESCRIPTION
**Pull request by jnatten**
- Defaults to filter statuses based only on current
- Adds `filter-other-statuses` flag to allow filtering both current and other.

### CODE REVIEW

File: src/main/scala/no/ndla/searchapi/controller/SearchController.scala [hunk 1]
```diff
@@ -161,6 +161,11 @@ trait SearchController {
          |Supported values are ${ArticleStatus.values.mkString(", ")}.""".stripMargin
     )
 
+    private val filterOtherStatuses =
+      Param[Option[Boolean]](
+        "filter-other-statuses",
```
Comment (gunnarvelle) [inline src/main/scala/no/ndla/searchapi/controller/SearchController.scala:166]: Er det bedre å kalle denne `include-other-statuses`?

Comment (jnatten) [reply src/main/scala/no/ndla/searchapi/controller/SearchController.scala:166]: Sure, har ingen sterke meninger :smile:

File: src/main/scala/no/ndla/searchapi/controller/SearchController.scala [hunk 2]
```diff
@@ -432,7 +432,7 @@ trait SearchController {
       val aggregatePaths = paramAsListOfString(this.aggregatePaths.paramName)
       val embedResource = paramOrNone(this.embedResource.paramName)
       val embedId = paramOrNone(this.embedId.paramName)
-      val filterOtherStatuses = booleanOrDefault(this.filterOtherStatuses.paramName, default = false)
+      val filterOtherStatuses = booleanOrDefault(this.includeOtherStatuses.paramName, default = false)
```
Comment (gunnarvelle) [inline src/main/scala/no/ndla/searchapi/controller/SearchController.scala:435]: Kunne endra navnet på denne også.

File: src/main/scala/no/ndla/searchapi/service/search/MultiDraftSearchService.scala [hunk 3]
```diff
@@ -216,13 +216,16 @@ trait MultiDraftSearchService {
       ).flatten
     }
 
-    private def draftStatusFilter(statuses: Seq[draft.ArticleStatus.Value]): Some[BoolQuery] = {
-      val draftStatuses = Seq("draftStatus.current", "draftStatus.other")
+    private def draftStatusFilter(statuses: Seq[draft.ArticleStatus.Value], filterOthers: Boolean): Some[BoolQuery] = {
```
Comment (gunnarvelle) [inline src/main/scala/no/ndla/searchapi/service/search/MultiDraftSearchService.scala:219]: Samme her. includeOthers.

**Review by gunnarvelle — APPROVED** [review]

**Pull request merged**